replica/table.cc

```diff
@@ -437,6 +438,21 @@ void table::enable_off_strategy_trigger() {
 
 future<>
 table::do_add_sstable_and_update_cache(sstables::shared_sstable sst, sstables::offstrategy offstrategy) {
+    if (sst->requires_view_building()) {
+        _sstables_staging.insert(sst);
+
+        if (_sstables->contains(sst->generation())) {
+            co_return;
+        }
+
+        // clone staging sstables so their content may be compacted while
+        // views are built.  When done, the hard-linked copy in the staging
+        // subsirectory will be simply unlinked.
+        co_await sst->create_links(dir());
+        sst = make_sstable(dir(), sst->generation(), sst->get_version(), sst->get_format());
+        co_await sst->load();
```

Comment:
By the way, I think we can avoid the resource duplication by introducing sstable::clone_at(new_dir).

please look at: future<foreign_sstable_open_info> sstable::get_open_info(). we can't reuse the file descriptors as cloned version will be at a different directory, but we don't have to copy in-memory metadata as files are immutable, therefore avoiding the 100% space overhead.

Reply:
Implementation could be something as follow:

```
future<shared_sstable>
sstable::clone_at(const sstring new_dir) noexcept {
    co_await create_links(new_dir);
    auto cloned_sst = _manager.make_sstable(_schema, new_dir, _generation, _version, _format);
    co_await cloned_sst->load(co_await get_open_info());
    co_return cloned_sst;
}
```

so we reduce read amplification and memory overhead, as we won't have to reload components and store in-memory metadata redundantly

Reply:
Sounds reasonable. I can take a stab at it.

Reply:
thanks for addressing this.

Reply:
done